sinks/external.go

```diff
@@ -43,18 +43,19 @@ func NewExternalSinkManager(externalSinks []sink_api.ExternalSink) (ExternalSink
 	for _, supported := range supportedMetrics {
 		descriptors = append(descriptors, supported.MetricDescriptor)
 	}
+	return descriptors
+}
 
-	for _, externalSink := range externalSinks {
-		err := externalSink.Register(descriptors)
-		if err != nil {
-			return nil, err
-		}
+// NewExternalSinkManager returns an external sink manager that will manage pushing data to all
+// the sinks in 'externalSinks', which is a map of sink name to ExternalSink object.
+func NewExternalSinkManager(externalSinks []sink_api.ExternalSink) (ExternalSinkManager, error) {
```

Comment:
Then we'd have to change lots more, like `externalSinkManager` to `sinkManager`. This goes hand in hand with "don't know the difference between Sink and ExternalSink".